### PR TITLE
Adding a merge checklist to release PRs

### PR DESCRIPTION
**Pull request by jonrohan**
This adds a pre-merge checklist to our Release PRs. I took this workflow and template from primer/react https://github.com/primer/react/pull/2365#issuecomment-1256377199

I tried my best to edit for things we'll be interested in, but please suggest changes if I missed anything

### CODE REVIEW

File: .github/release_template.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  - [ ] Install the npm release candidate in `npm-workspaces/primer/`
+  - [ ] Use `bin/vendor-gem` to update `primer_view_components` gem
+  - [ ] Verify no new build errors appear
+  - [ ] Verify no new linting errors appear
+  - [ ] Manually test critical paths
+  - [ ] Manually test release-specific bugfixes and/or features work as described
+
+### 🤔 Sanity test
+
+- [ ] All bugfixes in this release have resolved their corrosponding issues
```
Comment (camertron) [inline .github/release_template.md:19]:
```suggestion
- [ ] All bugfixes in this release have resolved their corresponding issues
```

File: .github/release_template.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+- [ ] All new features in this release have been tested and verified as compatible with GitHub.com
+- [ ] No noticeable regressions have not been introduced as a result of changes in this release
```
Comment (camertron) [inline .github/release_template.md:21]:
```suggestion
- [ ] No noticeable regressions have been introduced as a result of changes in this release
```

**Review by camertron — APPROVED** [review]
Are we sure we need something like this? I don't really have a problem with it but wonder if we need it for every PR.

**Comment by jonrohan** [comment]
> I don't really have a problem with it but wonder if we need it for every PR.

It should only run on the Release PRs like this one https://github.com/primer/view_components/pull/1433

**Review by langermank — APPROVED** [review]
Looks great!

**Pull request merged**